baseplate/lib/experiments/__init__.py

```diff
@@ -54,14 +91,18 @@ def __init__(
         timeout: Optional[float] = None,
         backoff: Optional[float] = None,
     ):
-        self._filewatcher = FileWatcher(path, json.load, timeout=timeout, backoff=backoff)
+        self._experiments_global_cache: ExperimentsGlobalCache = ExperimentsGlobalCache(
+            FileWatcherWithUpdatedFlag(path, json.load, timeout=timeout, backoff=backoff)
+        )
         self._event_logger = event_logger
 
     def make_object_for_context(self, name: str, span: Span) -> "Experiments":
+        cfg_data, global_cache = self._experiments_global_cache.get_cfg_and_global_cache()
```

Comment:
Ooo. So I was originally worried this would mean that we're no longer lazily loading the experiment config only when the current request actually tries to use experiments, but since Baseplate.py 1.0 `make_object_for_context` is only called lazily on use so this is fine and actually makes things nice and clean!

Do we actually need `ExperimentsGlobalCache`? Can we just do that logic right here in `make_object_for_context`?

Reply:
> Do we actually need ExperimentsGlobalCache? Can we just do that logic right here in make_object_for_context?

`+1`